### PR TITLE
[13.0] [MIG] stock_reserve_sale: Migration to v13.0

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT30810

This module depends on:
- [x]  #1202 

please @pedrobaeza @victoralmau review this

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please rebase when the other gets merged.

File: stock_reserve_sale/model/sale.py [hunk 1]
```diff
@@ -45,53 +28,100 @@ def _stock_reservation(self):
...
+            body += "{} → {}</div>".format(
+                old_vals["product_uom_qty"], float(new_vals["product_uom_qty"]),
+            )
+        body += "<br/>"
+        return body
+
+    def unlink(self):
+        for order in self:
+            if order.has_stock_reservation:
+                raise Warning(
```
Comment (rousseldenis) [inline stock_reserve_sale/model/sale.py]: Please use UserError as Warning is deprecated.

File: stock_reserve_sale/model/sale.py [hunk 2]
```diff
@@ -108,23 +138,21 @@ def _get_line_rule(self):
                 ("route_id", "in", wh_route_ids),
             ]
 
-            rules = ProcurementRule.search(domain, order="route_sequence, sequence")
+            rules = StockRule.search(domain, order="route_sequence, sequence")
 
         if rules:
             return rules[0]
```
Comment (rousseldenis) [inline stock_reserve_sale/model/sale.py]: You can use directly:
```suggestion
            return fields.first(rules)
```

File: stock_reserve_sale/model/sale.py [hunk 3]
```diff
@@ -258,7 +227,19 @@ def write(self, vals):
...
                         "product_uom_qty": line.product_uom_qty,
-                        "product_uos_qty": line.product_uos_qty,
                     }
                 )
         return res
+
+    def unlink(self):
+        for line in self:
+            if line.reservation_ids:
+                raise Warning(
```
Comment (rousseldenis) [inline stock_reserve_sale/model/sale.py]: Same as above

File: stock_reserve_sale/tests/test_stock_reserve_sale.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
+# Copyright 2021 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo.exceptions import Warning
+from odoo.tests import Form, common
+
+
+class TestStockReserveSale(common.TransactionCase):
```
Comment (rousseldenis) [inline stock_reserve_sale/tests/test_stock_reserve_sale.py]: Using SavepointCase is better

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Great to see some excavating this old module!

Code review.

File: stock_reserve_sale/tests/test_stock_reserve_sale.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    def setUp(self):
+        super().setUp()
+        partner_form = Form(self.env["res.partner"])
+        partner_form.name = "Test partner"
```
Comment (victoralmau) [inline stock_reserve_sale/tests/test_stock_reserve_sale.py:11]:
```suggestion
        partner_form.name = "Test partner"
        partner_form.country_id = self.env.ref("base.es")
```

**Review by victoralmau — CHANGES_REQUESTED** [review]

**Comment by CarlosRoca13** [comment]
@rousseldenis @victoralmau changes done

**Review by rousseldenis — APPROVED** [review]
Code review

**Review by victoralmau — APPROVED** [review]

**Pull request merged**